incrementer/prometheus.go

```diff
@@ -0,0 +1,106 @@
+package incrementer
+
+import (
+	"github.com/hellofresh/stats-go/bucket"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Prometheus struct is Incrementer interface implementation that writes all metrics to Prometheus
+type Prometheus struct {
+	counter *prometheus.CounterVec
+}
+
+// NewPrometheus creates new prometheus incrementer instance
+func NewPrometheus() *Prometheus {
+	return &Prometheus{counter: nil}
+}
+
+// Increment increments metric in prometheus
+func (i *Prometheus) Increment(metric string) {
+	if i.counter == nil {
+		i.counter = prometheus.NewCounterVec(
```

Comment:
why not created it in the constructor?

Reply:
Because new counter is created only when we call it first time with defined metric name